server/events/project_command_builder.go

```diff
@@ -415,3 +428,35 @@ func (p *DefaultProjectCommandBuilder) escapeArgs(args []string) []string {
 	}
 	return escaped
 }
+
+// Extracts required_version from Terraform configuration.
+// Returns nil if unable to determine version from configuation, check warning log for clarification.
+func (p *DefaultProjectCommandBuilder) getTfVersion(ctx *CommandContext, absProjDir string) *version.Version {
+	module, diags := tfconfig.LoadModule(absProjDir)
+	if diags.HasErrors() {
+		ctx.Log.Warn(diags.Error())
+		return nil
+	}
+
+	if len(module.RequiredCore) != 1 {
+		ctx.Log.Warn("cannot determine required core from terraform configuration, detected %d possibilities.", len(module.RequiredCore))
+		return nil
+	}
+
+	ctx.Log.Debug("detected core: \"%q\".", module.RequiredCore[0])
+	re := regexp.MustCompile(`^=?\s*([^\s]+)\s*$`)
+	matched := re.FindStringSubmatch(module.RequiredCore[0])
+	if len(matched) == 0 {
+		ctx.Log.Warn("did not specify exact version in terraform configuration.")
```

Comment:
I would unify log levels here or switch those message priorities as the fact that a version was detected and is being use is IMHO more important than the fact that we fallback to defaults because no version was detected.

Reply:
Hmmm you make sense. I will update my PR.

Reply:
I added some more logs and unified as you suggested.